### PR TITLE
Updated order-api for starlims mip-dna-json

### LINKED ISSUE #1190: Updated order-api for starlims mip-dna-json
## Description

At a meeting 2021-06-04 it was decided:
To make changes to the mip-dna-json-api in the Orderportal. The contact person is PG at our end. The new files will be sent to PG in September for testing.

## Suggested solution

1. Change cg/tests/fixtures/orderforms/mip-json.json with new spec.
2. Change tests for new spec.
3. Fix implementation

## This can be closed when

- [x] change synopsis: [str] -> synopsis: str
- [x] new phenotype_groups: [str]
- [x] new subject_id: str (maybe change also for mip-rna 'for-sample'?)
- [ ] new volume: str
- [x] phenotype groups are shown in cgweb
- [x] synopsis is treated as text in cgweb
- [ ] synopsis: str is sent to scout
- [ ] phenotype_groups is sent to scout
- [ ] (subject_id is sent to scout?)

## Blocked by

If the api can handle the old fixture as well as the new then it is OK to deploy it to production

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
## Description
This PR resolves #1190 

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### Affected workflows:
- Balsamic
- MIP-DNA
- MIP-RNA

### New/changed:
- phenotype_groups (new)
- subject_id (new)
- synopsis (changed from list to string)

### Test cases:
- data parsed from json (all pipelines)
- data shown in Orderportal
- data saved in db (all pipelines)
- data sent to Scout (all pipelines)

### How to test parsing and showing in Orderportal and saving in database
- [x] do upload Order-json

### Expected test outcome
Check that data was saved in the database
- MIP-DNA
  - [x] phenotype-groups
  - [x] subject_id
  - [x] synopsis
- [x] Balsamic
  - [x] phenotype-groups
  - [x] phenotype-terms
  - [x] subject_id
  - [x] synopsis
  - [x] cohorts
- [x] MIP-RNA
  - [x] phenotype-groups
  - [x] phenotype-terms
  - [x] subject_id
  - [x] synopsis 
  - [x] cohorts

### How to test uploading to Scout
- [x] set cohorts and synopsis on case justhusky
- [x] set subject_id, phenotype_terms, phenotype_groups on sample(s) of justhusky
- [x] `cg upload scout --print justhusky`

### Expected test outcome
Check that data was uploaded for:
- [x] phenotype-groups
- [x] phenotype-terms
- [x] subject_id
- [x] synopsis 
- [x] cohorts

- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] code approved by BS
- [x] tests executed by PG
- [x] "Merge and deploy" approved by PG
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [x] Document in changelog
- [ ] ~Bump version~
- [ ] Deploy this branch on clinical-db
- [ ] Deploy this branch on Hasta
- [ ] Inform to Starlims contact & ES

### CODE REVIEW

**Comment by patrikgrenfeldt** [comment]
Test result mip-dna order: 
image
🆗 phenotype groups an subject_id saved in db

**Comment by patrikgrenfeldt** [comment]
Test result upload scout:
image
image
image
image
image
🆗 synopsis, phenotype_groups, phenotype_terms, cohorts and subject_id present

**Comment by patrikgrenfeldt** [comment]
Test result ordering mip-dna, balsamic, mip-rna:
image
image
🆗 synopsis, phenotype_groups, phenotype_terms, cohorts, subject_id present

**Review by barrystokman — APPROVED** [review]
👍🏻 

Looks good, had only a very tiny remark :)

File: cg/models/orders/json_sample.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import typing
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/models/orders/json_sample.py]: removed now

**Comment by patrikgrenfeldt** [comment]
Deployed on Hasta:
```
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 22.19.4
    Uninstalling cg-22.19.4:
      Successfully uninstalled cg-22.19.4
Successfully installed cg-22.19.7
```

Deployed on Clinical-db:
´´´
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 22.19.5
    Uninstalling cg-22.19.5:
      Successfully uninstalled cg-22.19.5
Successfully installed cg-22.19.7
```

**Comment by patrikgrenfeldt** [comment]
@emmser We have deployed this new version of the orderportal and the clinical-api

**Pull request merged**